src/context/user_context.js

```diff
@@ -1,13 +1,52 @@
 import React, { useContext, useEffect, useState } from 'react'
+import axios, { AxiosError } from "axios";
+import { user_url } from '../utils/constants'
 
+async function fetchCurrentUser() {
+  const url = `${user_url}/showMe`;
+  return axios
+    .get(url, { withCredentials: true })
+    .then((response) => {
+      console.log("fetchCurrentUser response: ", response);
+      return response?.data?.user;
+    })
+    .catch((error) => {
+      const errorPayload =
+        error instanceof AxiosError ? error?.response?.data : error;
+      console.error(errorPayload);
+    });
+}
```

Comment:
I put the function outside of the component partially to make it easier to read; and also because that means React won't have to consider the function as part of the component every time it re-renders the `UserProvider` component (this is usually just a very small optimization, but in very large React apps you may notice that it is better practice to define static functions _outside_ of the component like this)